### PR TITLE
TOOLS/VFS: Infrastructure for FUSE-based monitoring file-system

### PR DESCRIPTION
**Pull request by rakhmets**
Cherry-picked a set of commits from master to int3 to be able to port VFS-related methods to the branch.
Major part of changes was cherry-picked w/o conflicts, but minor conflicts were fixed in a couple of commits.

### CODE REVIEW

**Review by brminich — APPROVED** [review]

**Comment by rakhmets** [comment]
Run gtests locally. No regressions were found.

**Comment by rakhmets** [comment]
@yosefe could you please check this PR

File: src/ucs/config/global_opts.c [hunk 1]
```diff
@@ -184,6 +185,10 @@ static ucs_config_field_t ucs_global_opts_table[] = {
 
 #endif
 
+ {"VFS_ENABLE", "y",
```
Comment (yosefe) [inline src/ucs/config/global_opts.c]: let's set it to "n" here to be on the safe side. at least until porting all fuse commits from master, including the fix for fork() and hang at exit.

File: test/gtest/common/gtest-all.cc [hunk 2]
```diff
@@ -7489,7 +7489,7 @@ void StackLowerThanAddress(const void* ptr, bool* result) {
 }
 
 bool StackGrowsDown() {
-  int dummy;
+  int dummy = 0;
```
Comment (yosefe) [inline test/gtest/common/gtest-all.cc:7492]: why needed?

File: test/gtest/ucs/test_pgtable.cc [hunk 3]
```diff
@@ -524,7 +524,9 @@ class test_pgtable_perf : public test_pgtable {
 
 private:
     struct region_comparator {
-        bool operator()(ucs_pgt_region_t* region1, ucs_pgt_region_t* region2) {
+        bool
+        operator()(ucs_pgt_region_t *region1, ucs_pgt_region_t *region2) const
```
Comment (yosefe) [inline test/gtest/ucs/test_pgtable.cc:528]: why needed?

File: test/gtest/common/gtest-all.cc (hunk 2 above)
Comment (rakhmets) [reply test/gtest/common/gtest-all.cc:7492]: This change was applied along with cherry pick of [UCS/SOCK: Fix RPM build with gcc11 on fedora34](https://github.com/openucx/ucx/commit/81b169f2b562c835db41f58af7016fe3beec3459)

File: test/gtest/ucs/test_pgtable.cc (hunk 3 above)
Comment (rakhmets) [reply test/gtest/ucs/test_pgtable.cc:528]: This change was applied along with cherry pick of [CONFIG/VFS/TEST: Fix compilation on gcc11](https://github.com/openucx/ucx/commit/df43abdd8e4f2773f2f358cd70d111fcc3e05a63)

**Pull request closed without merging**